### PR TITLE
file and python dependents: add missing dependencies

### PR DESCRIPTION
**Pull request by sethrj**
- Mark redundant python-magic package
- Add dependencies on the `file` package: note that these libraries import `libmagic.{so,dylib,dll}` at runtime rather than calling the `file` command, so not all external installations of `file` will work here!
- Add compressor dependencies to `file`; previously it was linking against system compression libraries rather than spack ones.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-magic/package.py [hunk 1]
```diff
@@ -1,18 +0,0 @@
-# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
-# Spack Project Developers. See the top-level COPYRIGHT file for details.
-#
-# SPDX-License-Identifier: (Apache-2.0 OR MIT)
-
-from spack import *
-
-
-class PyMagic(PythonPackage):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-magic/package.py:9]: I agree that this is a duplicate package, but we should first deprecate it and remove it in a future release. Just add `deprecated=True` to the version.

File: var/spack/repos/builtin/packages/libmagic/package.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    NOTE: some package managers delete the resulting "file" command to avoid
+    installing the system utility, but we currently do not. Note also that some
+    system-installed file commands do *not* come paired with shared-object
+    libmagic libraries that are necessary to load python-based packages
+    (py-python-magic, py-filemagic).
+    """
+
+    homepage = "https://www.darwinsys.com/file/"
+    url      = "https://astron.com/pub/file/file-5.40.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libmagic/package.py]: Isn't this a duplicate of our `file` package?

File: var/spack/repos/builtin/packages/py-magic/package.py (hunk 1 above)
Comment (sethrj) [reply var/spack/repos/builtin/packages/py-magic/package.py:9]: Fair enough!

File: var/spack/repos/builtin/packages/libmagic/package.py (hunk 2 above)
Comment (sethrj) [reply var/spack/repos/builtin/packages/libmagic/package.py]: Well darn, I should have checked that first! Yes, will fix.

File: var/spack/repos/builtin/packages/py-python-magic/package.py [hunk 3]
```diff
@@ -8,11 +8,16 @@
...
+    """A python wrapper for libmagic.
+
+    This project is named python-magic but imports as the module name "magic".
+    """
 
     homepage = "https://github.com/ahupp/python-magic"
     pypi = "python-magic/python-magic-0.4.15.tar.gz"
 
+    version('0.4.24', sha256='de800df9fb50f8ec5974761054a708af6e4246b03b4bdaee993f948947b0ebcf')
     version('0.4.15', sha256='f3765c0f582d2dfc72c15f3b5a82aecfae9498bd29ca840d72f37d7bd38bfcd5')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-python-magic/package.py:21]: Can you also add:
```python
depends_on('python@2.7.0:2.7,3.5:', type=('build', 'run'))
```
or whatever the magic syntax is these days?

File: var/spack/repos/builtin/packages/file/package.py [hunk 4]
```diff
@@ -15,15 +15,34 @@ class File(AutotoolsPackage):
...
         match = re.search(r'file-(\S+)', output)
         return match.group(1) if match else None
+
+    def configure_args(self):
+        args = [
+            "--disable-dependency-tracking",
+            "--enable-fsect-man5",
+            "--enable-zlib",
+            "--enable-bzlib",
+            "--enable-xzlib",
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/file/package.py:45]: If these are optional, we might want to add variants and make them off by default. Trying to minimize build times. But this isn't my package so if you're the maintainer you get to decide 😄

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/file/package.py (hunk 4 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/file/package.py:45]: some packages have a multivalues variant `compression=x,y,z` now btw.

Comment (sethrj) [reply var/spack/repos/builtin/packages/file/package.py:45]: Build times are quite small, and as I mentioned in the description the default was to turn them on and link against system libraries. I'm happy to review a PR for adding compression variants now that I'm maintainer ;)

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/file/package.py:45]: Yeah, we def don't want to link to system libs regardless.

**Pull request merged**